### ISSUE TITLE
Remove usage of Exception.message

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8098]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8098|
|Type|enhancement|
|Created|2015-11-07 15:59:02Z|
|Branch|https://github.com/twisted/twisted/tree/dotmessage-8098|

This is deprecated in Python 2.6, and we shouldn't use it.

<details><summary>Searchable metadata</summary>

```
trac-id__8098 8098
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_dotmessage_8098 branches/dotmessage-8098
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1446911942113925 1446911942113925
changetime__1447413490826983 1447413490826983
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Builds are looking good. Please review.

Please see the lack of the .message warnings in https://buildbot.twistedmatrix.com/builders/debian8-py2.7/builds/535/steps/select/logs/warnings .

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_3"></a>|@derwolfe set owner to @hawkowl|
|-|-|

Hi Hawkie,

1. I've compared the logs to trunk and all of the .message warnings look to have been addressed.
2. Using ``` .args[0] ``` is consistent with other code in Twisted and seems to be a better alternative than using ``` str(e) ```.

This looks good to land. Please merge.

Thanks!
Herrwolfe

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46178]) Merge dotmessage-8098: Remove usages of Exception.message from Twisted

Author: hawkowl
Reviewer: herrwolfe
Fixes: [#8098](https://github.com/twisted/twisted/issues/8098)
